### PR TITLE
Sampler array no longer necessary

### PR DESCRIPTION
**Pull request by Ceanze**
### Changes
* When using updateResoruce or writeDescriptorSet it is no longer necessary to have the same length for the sampler as the texture arrays. This is done since most of the time we are using the same sampler for all of the textures (nearest or linear).

### CODE REVIEW

File: LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp [hunk 1]
```diff
@@ -1100,22 +1100,21 @@ namespace LambdaEngine
...
 						m_PaintMaskTextures.PushBack(ResourceManager::GetTexture(GUID_TEXTURE_DEFAULT_MASK_MAP));
 						m_PaintMaskTextureViews.PushBack(ResourceManager::GetTextureView(GUID_TEXTURE_DEFAULT_MASK_MAP));
-						m_PaintMaskSamplers.PushBack(Sampler::GetNearestSampler());
 					}
 
 					m_PaintMaskTextures.PushBack(pTexture);
 					m_PaintMaskTextureViews.PushBack(pTextureView);
-					m_PaintMaskSamplers.PushBack(pNearestSampler); // In an ideal world we would only have one sampler instead of a list
-					// TODO: Update rendergraph to support only one sampler for several texture views
+					Sampler* sampler = Sampler::GetNearestSampler();
```
Comment (TheoBerlin) [inline LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp]: `pSampler`

File: LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp [hunk 2]
```diff
@@ -2140,15 +2139,16 @@ namespace LambdaEngine
 		if (needUpdate)
 		{
 			uint32 texturesExisting = m_CubeTextures.GetSize();
-			TArray<Sampler*> nearestSamplers(texturesExisting, Sampler::GetNearestSampler());
+			Sampler* nearestSampler = Sampler::GetNearestSampler();
```
Comment (TheoBerlin) [inline LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp]: `pNearestSampler`

File: LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp [hunk 3]
```diff
@@ -2289,28 +2289,31 @@ namespace LambdaEngine
 
 			m_pRenderGraph->UpdateResource(&resourceUpdateDesc);
 
-			TArray<Sampler*> linearSamplers(m_AlbedoMaps.GetSize(), Sampler::GetLinearSampler());
+			Sampler* linearSamplers = Sampler::GetLinearSampler();
```
Comment (TheoBerlin) [inline LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp]: `pLinearSampler`

**Review by TheoBerlin — CHANGES_REQUESTED** [review]
Only found missing `p` prefixes on pointer variables. Also, it looks like you can either have a unique sampler for every texture, or only have one sampler for each texture. But having 2 samplers for 3 textures isn't supported. Is that correct, and if so, is it intentional?

**Comment by Ceanze** [comment]
> Only found missing `p` prefixes on pointer variables. Also, it looks like you can either have a unique sampler for every texture, or only have one sampler for each texture. But having 2 samplers for 3 textures isn't supported. Is that correct, and if so, is it intentional?

It is indeed correct that having samplerCount > 1 and samplerCount != textureCount is unsupported. This is because I believe it is unclear on which sampler is used for which texture in that situation.
Also, in the current implemenation there are not situations where that is applied.

File: LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp (hunk 3 above)
Comment (Ceanze) [reply LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp]: Fixed

File: LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp (hunk 1 above)
Comment (Ceanze) [reply LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp]: Fixed

File: LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp (hunk 2 above)
Comment (Ceanze) [reply LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp]: Fixed

**Review by TheoBerlin — APPROVED** [review]

**Pull request merged**